examples/hello_world/src/lib.rs

```diff
@@ -31,27 +31,23 @@ use std::io::Write;
 #[derive(OakNode)]
 struct Node;
 
-impl Node {
-    fn say_hello(&self, req: &HelloRequest) -> HelloResponse {
-        let mut res = HelloResponse::new();
-        info!("Say hello to {}", req.greeting);
-        res.reply = format!("HELLO {}!", req.greeting);
-        res
-    }
+// TODO: Generate this code via a macro or code generation (e.g. a protoc plugin).
+trait HelloWorldNode {
+    fn say_hello(&self, req: &HelloRequest) -> HelloResponse;
 }
 
+// TODO: Generate this code via a macro or code generation (e.g. a protoc plugin).
 impl oak::Node for Node {
     fn new() -> Self {
         oak_log::init(log::Level::Debug).unwrap();
         Node
     }
     fn invoke(&mut self, grpc_method_name: &str, grpc_channel: &mut oak::Channel) {
         let mut logging_channel = oak::logging_channel();
-        // TODO: Generate this code via a macro or code generation (e.g. a protoc plugin).
         match grpc_method_name {
             "/oak.examples.hello_world.HelloWorld/SayHello" => {
                 let req = protobuf::parse_from_reader(grpc_channel).unwrap();
-                let res = self.say_hello(&req);
+                let res = (self as &mut HelloWorldNode).say_hello(&req);
```

Comment:
I'm surprised this is needed, could you confirm it does not compile without it?

Reply:
Heh... it does compile without it. I had a chat with @daviddrysdale about this yesterday. While having a trait for `say_hello` is nice, there is nothing statically enforcing the user to implement it. So users can say something like `impl Node` which basically means you have two ways to do the same thing. This way, we actually force the user to implement the trait, which is not giving us any run-time advantages, but we believe it creates a clear and self documenting interface. Happy to revisit this if you think this is overkill.